### PR TITLE
fix: exec hang, exec trailing options

### LINKED ISSUE #15: melos exec hang
when i run `melos exec -c=1 -- flutter pub run build_runner build --delete-conflicting-outputs` it hang after the completion of the first package.

from what i have debug it seems `await processStdoutCompleter.future` line [218](https://github.com/invertase/melos/blob/master/lib/src/common/utils.dart#L218) never complete but i didn't understand why for now :/

EDIT: After more investigation it seems the exit command pushed in stdin after the command is eat by the running program.

EDIT2: Also i'm not sure why you are starting a shell yourself since the `runInShell` parameter on `Process.start` already do it for you ?

### PR DESCRIPTION
**Pull request by 2ZeroSix**
closes #15
closes #16
inspired by #16

- pass script source directly to shell using `-c` option
- ignore trailing options in exec command (exec tried to parse all available options including those passed to the child script)
  same approach is used in `pub run` https://github.com/dart-lang/pub/blob/291705ca0b9632cb945dd39493dd5b9db41b897a/lib/src/command/run.dart#L27

### CODE REVIEW

**Review by Salakar — APPROVED** [review]

File: packages/melos/lib/src/common/utils.dart [hunk 1]
```diff
@@ -213,25 +213,19 @@ Future<int> startProcess(
     return _arg;
   }).where((element) => element != null);
 
+  final scriptSource = filteredArgs.join(' ');
   final execProcess = await Process.start(
     executable,
-    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : [],
+    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : ['-c', scriptSource],
```
Comment (Salakar) [inline packages/melos/lib/src/common/utils.dart:219]:
```suggestion
    currentPlatform.isWindows ? ['/C', '%MELOS_SCRIPT%'] : ['-c', r'$MELOS_SCRIPT'],
```

Wouldn't this work? Keeps it consistent with how Windows does it too then

Comment (2ZeroSix) [reply packages/melos/lib/src/common/utils.dart]: It causes wrong variable expansion, you can see in test:
`bin/sh: 1: "dart: not found`

single argument `"dart analyze . --fatal-infos"`
becomes `"dart` `analyze` `.` `--fatal-infos"`

Comment (2ZeroSix) [reply packages/melos/lib/src/common/utils.dart]: @Salakar ^

Comment (Salakar) [reply packages/melos/lib/src/common/utils.dart]: I see now, will revert, one moment

Comment (Salakar) [reply packages/melos/lib/src/common/utils.dart]: What's your thoughts on `['-c', r'eval "$MELOS_SCRIPT"']` - since this does work

Comment (Salakar) [reply packages/melos/lib/src/common/utils.dart]: Have tested on Melos and FlutterFire and everything seems to be working with the above so will merge - I can change in another PR if needed. Thanks for your help :)

**Pull request merged**